File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/inferenceClassification/InferenceClassificationOverride.java
package graphjavav4.utilities.users.inferenceClassification;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import graphjavav4.utilities.users.EmailAddress;
import graphjavav4.utilities.users.Entity;
import graphjavav4.utilities.users.InferenceClassificationType;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
public class InferenceClassificationOverride extends Entity implements Parsable {
    @javax.annotation.Nullable
    public InferenceClassificationType classifyAs;
    @javax.annotation.Nullable
    public EmailAddress senderEmailAddress;
    /**
     * The deserialization information for the current model
     * @return a Map<String, BiConsumer<T, ParseNode>>
     */
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getFieldDeserializers() {
        return new HashMap<>(super.getFieldDeserializers()) {{
            this.put("classifyAs", (o, n) -> { ((InferenceClassificationOverride)o).classifyAs = n.getEnumValue(InferenceClassificationType.class); });
            this.put("senderEmailAddress", (o, n) -> { ((InferenceClassificationOverride)o).senderEmailAddress = n.getObjectValue(EmailAddress.class); });
        }};
    }
    /**
     * Serialiazes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @return a void
     */
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        super.serialize(writer);
        writer.writeEnumValue("classifyAs", classifyAs);
        writer.writeObjectValue("senderEmailAddress", senderEmailAddress);
    }
}

File: petstore/ruby/graphrubyv4/users/inference_classification/inference_classification_override.rb
require 'microsoft_kiota_abstractions'
require_relative '../entity'
require_relative '../users'
require_relative './inference_classification'

module Graphrubyv4::Users::InferenceClassification
    class InferenceClassificationOverride < Graphrubyv4::Users::Entity
        include MicrosoftKiotaAbstractions::Parsable
        @classify_as
        @sender_email_address
        ## 
        ## Gets the classifyAs property value. 
        ## @return a inference_classification_type
        ## 
        def  classify_as
            return @classify_as
        end
        ## 
        ## Gets the senderEmailAddress property value. 
        ## @return a email_address
        ## 
        def  sender_email_address
            return @sender_email_address
        end
        ## 
        ## The deserialization information for the current model
        ## @return a i_dictionary
        ## 
        def get_field_deserializers() 
            return super.merge({
                "classifyAs" => lambda {|o, n| o.classify_as = n.get_enum_value(Graphrubyv4::Users::InferenceClassificationType) },
                "senderEmailAddress" => lambda {|o, n| o.sender_email_address = n.get_object_value(Graphrubyv4::Users::EmailAddress) },
            })
        end
        ## 
        ## Serializes information the current object
        ## @param writer Serialization writer to use to serialize this model
        ## @return a void
        ## 
        def serialize(writer) 
            super
            writer.write_enum_value("classifyAs", @classify_as)
            writer.write_object_value("senderEmailAddress", @sender_email_address)
        end
        ## 
        ## Sets the classifyAs property value. 
        ## @param value Value to set for the classifyAs property.
        ## @return a void
        ## 
        def  classify_as=(classifyAs)
            @classify_as = classifyAs
        end
        ## 
        ## Sets the senderEmailAddress property value. 
        ## @param value Value to set for the senderEmailAddress property.
        ## @return a void
        ## 
        def  sender_email_address=(senderEmailAddress)
            @sender_email_address = senderEmailAddress
        end
    end
end
